### ISSUE TITLE
Different html without any changes in content

### ISSUE THREAD

**Issue by stopdesign** [gohugoio/hugo#11749]
Hugo renders pages differently from time to time.

For a minimal reproducible example I have created an empty project with `hugo new site` and one content/index.md file.

I have tried hugo-book and archie themes, same result.

```
$ hugo version
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended darwin/arm64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew
```

```
$ uname -a
Darwin localhost.local 21.6.0 Darwin Kernel Version 21.6.0: Wed Oct  4 23:56:02 PDT 2023; root:xnu-8020.240.18.704.15~1/RELEASE_ARM64_T6000 arm64
```

```$ cat hugo.toml
baseURL = 'https://example.org/'
languageCode = 'en-us'
title = 'My Hugo Site 6798'
theme = 'hugo-book'
```

```
$ cat content/index.md
+++
title = "index page"
+++

index.md content:

asdf
lllxxllll
```

Let’s say that immediately after launching `hugo server`, the site looks as expected. After several updates (ctrl-s without changes), the index.md content disappears from the main page. After a few more saves, the content returns.

The error happens in `hugo server` and `hugo`.
Here are two consecutive runs that produce different sizes of index.html.

```
$ hugo --logLevel debug ; ls -al public | grep "index.html"
Start building sites …
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended darwin/arm64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew

INFO  copy static: syncing static files to /
INFO  build: running step process duration 74.375µs
INFO  build: running step assemble duration 90.084µs
INFO  build: running step render duration 12.414167ms
INFO  build: running step postProcess duration 2.916µs

                   | EN
-------------------+-----
  Pages            |  8
  Paginator pages  |  0
  Non-page files   |  0
  Static files     | 78
  Processed images |  0
  Aliases          |  2
  Sitemaps         |  1
  Cleaned          |  0

Total in 21 ms
-rw-r--r--  1 gregory staff    3695 Nov 27 13:03 index.html
```

```
$ hugo --logLevel debug ; ls -al public | grep "index.html"
Start building sites …
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended darwin/arm64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew

INFO  copy static: syncing static files to /
INFO  build: running step process duration 73.833µs
INFO  build: running step assemble duration 84.334µs
INFO  build: running step render duration 12.293583ms
INFO  build: running step postProcess duration 4.083µs

                   | EN
-------------------+-----
  Pages            |  8
  Paginator pages  |  0
  Non-page files   |  0
  Static files     | 78
  Processed images |  0
  Aliases          |  2
  Sitemaps         |  1
  Cleaned          |  0

Total in 21 ms
-rw-r--r--  1 gregory staff    3742 Nov 27 13:03 index.html
```

**Comment by jmooring** [comment]
Your home page should be _index.md, not index.md.

**Comment by jmooring** [comment]
In the future, please use the forum (https://discourse.gohugo.io/) for questions and troubleshooting. We prefer to use GitHub for verified bugs and vetted enhancements. Thanks.

**Comment by stopdesign** [comment]
It was posted as an issue deliberately. I don't have much experience with Hugo. But I think non-deterministic disappearing pages **without any warnings** in debug mode is a bug or a bad design.

Look at this from my perspective. I tried this platform for a few days, and everything was ok. The next day, for **some unknown reason** pages start disappearing randomly, links become broken, and so on. Non-deterministic behavior is the worst issue. I looked at the documentation but did not realize it was the wrong file name.

I assume this "_index" vs "index" thing is a common issue. The index page is a well-known concept for a static websites. Hugo violates this concept without a clear warning (the documentation doesn't explain it clearly, at least for my level of hugo-related knowlage https://gohugo.io/content-management/organization/#index-pages-_indexmd).

Anyway... That was a perspective of a web-developer with 15 years of experience with HTML and 3 days with Hugo. I understand what went wrong. And after looking at discourse.gohugo.io I still think the issue should be addressed somehow.
